PINFuture/Classes/PINFuturePending.m

```diff
@@ -0,0 +1,65 @@
+//
+//  PINFuturePending.m
+//  Pods
+//
+//  Created by Ray Cho on 2/20/20.
+//
+
+#import "PINFuturePending.h"
+
+@interface PINFuturePending ()
+@property (nonatomic, strong) PINFuture<id> *pendingFuture;
+@property (nonatomic, copy) void (^resolve)(id value);
+@property (nonatomic, copy) void (^reject)(NSError *error);
+@end
+
+@implementation PINFuturePending
+
+- (instancetype)init
+{
+    self = [super init];
+    if (self) {
+        _pendingFuture = [PINFuture withBlock:^(void (^_Nonnull resolve)(id), void (^_Nonnull reject)(NSError *_Nonnull)) {
+            self.resolve = resolve;
+            self.reject = reject;
+        }];
+    }
+    return self;
+}
+
+- (PINFuture<id> *)future
+{
+    return self.pendingFuture;
+}
+
+- (void)fulfillWithValue:(id)value
+{
+    NSParameterAssert(value != nil);
+    NSParameterAssert([value isKindOfClass:[NSError class]] == NO);
+
+    if (self.resolve) {
+        self.resolve(value);
+    } else {
+        NSAssert(NO, @"Underlying future has already been resolved.");
```

Comment:
It might be nicer to write this above the `if` (rather than in an `else` block) as:

```
NSAssert(self.resolve, @"Underlying future has already been resolved.");
```

... although the way you have it makes it easier to set a breakpoint on the error condition.

Reply:
That sounds good. I just personally like having `else` cases to be clearer that we are handling all cases